dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -311,7 +311,7 @@ DynamicDataImpl::SingleValue::SingleValue(ACE_OutputCDR::from_wchar value)
 {}
 
 DynamicDataImpl::SingleValue::SingleValue(const CORBA::WChar* wstr)
-  : kind_(TK_STRING16), wstr_(ACE_OS::strdup(wstr))
+  : kind_(TK_STRING16), active_(&wstr_), wstr_(ACE_OS::strdup(wstr))
```

Comment:
```suggestion
  : kind_(TK_STRING16), active_(0), wstr_(ACE_OS::strdup(wstr))
```
Similar to the narrow string case.